pkg/kernel/rule.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright (c) 2025 OpenInfra Foundation Europe
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kernel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/nordix/meridio/pkg/log"
+	"github.com/nordix/meridio/pkg/utils"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+type RuleMatchFunc func(*netlink.Rule) bool
+
+// SkipBuiltInRules returns true if the given netlink.Rule's table is not one of the
+// standard built-in or unspec routing tables (local, main, default).
+func SkipBuiltInRules(rule *netlink.Rule) bool {
+	switch rule.Table {
+	case unix.RT_TABLE_UNSPEC:
+		fallthrough
+	case unix.RT_TABLE_COMPAT:
+		fallthrough
+	case unix.RT_TABLE_LOCAL:
+		fallthrough
+	case unix.RT_TABLE_MAIN:
+		fallthrough
+	case unix.RT_TABLE_DEFAULT:
+		return false
+	default:
+		return true
+	}
+}
+
+// FlushRules lists and deletes IP rules for the specified family that match the
+// provided RuleMatchFunc.
+// If matchFn is nil, all rules for the given family will be attempted to be deleted.
+// Errors encountered during rule deletion are accumulated and returned.
+func FlushRules(ctx context.Context, family int, matchFn RuleMatchFunc) error {
+	rules, err := netlink.RuleList(family)
```

Comment:
`RuleListFiltered` could be used to now have list all rules. You need to flush only rules with table id to 1 if I understand correctly.

Reply:
You are right. I had this change before deciding on using a common routing table.
On the other hand RuleListFiltered also simply dumps all the rules and then applies the filter in any to determine the return set. Yet, I will change the order of the two main commits and use RuleListFiltered as it would be more fitting.